nautobot/docs/plugins/development.md

```diff
@@ -149,6 +149,7 @@ The configurable attributes for a `PluginConfig` are listed below in alphabetica
 | `jobs` | The dotted path to the list of Job classes (default: `jobs.jobs`) |
 | `max_version` | Maximum version of Nautobot with which the plugin is compatible |
 | `menu_items` | The dotted path to the list of menu items provided by the plugin (default: `navigation.menu_items`) |
+| `menu_tabs` | The dotted path to the list of menu tab items provided by the plugin (default: `navigation.menu_tabs`) |
```

Comment:
Rather than adding a new, separate item to the plugin declaration, can we make it so that `menu_items` can be defined in either the existing style (`PluginMenu*`) or the new style (`NavMenu*`)? I think that would be a bit more elegant. It feels like it should be possible to do something to automatically translate the old style into the new style in the core code - something like:

```python
if isinstance(menu_items[0], PluginMenuItem):
    # translate old-style plugin menu definitions into the new nav-menu items and buttons
    new_menu_items = []
    new_menu_item_weight = 100
    for menu_item in menu_items:        
        new_menu_button_weight = 100
        new_menu_buttons = []
        for button in menu_item.buttons:
            new_menu_buttons.append(NavMenuButton(
                link=button.link, title=button.title, icon_class=button.icon_class, permissions=button.permissions, weight=new_menu_button_weight
            ))
            new_menu_button_weight += 100

        new_menu_items.append(NavMenuItem(
            link=menu_item.link, link_text=menu_item.link_text, permissions=menu_item.permissions, weight=new_menu_item_weight, buttons=new_menu_buttons
        ))
        new_menu_item_weight += 100

    # wrap bare item/button list into the default "Plugins" menu tab and appropriate grouping
    menu_items = (
        NavMenuTab(name="Plugins", weight=..., groups=(
            NavMenuGroup(name=pluginconfig.verbose_name, weight=..., items=new_menu_items),
        ),
    )
```


Basically, let's think about the current "two approaches" situation as something temporary only - any groundwork we can lay now towards converging the two towards a single solution based around the NavMenu code is worth doing.